lib/js_routes/instance.rb

```diff
@@ -24,9 +24,14 @@ def initialize(**options)
     sig {returns(String)}
     def generate
       # Ensure routes are loaded. If they're not, load them.
+
       application = T.unsafe(self.application)
-      if named_routes.empty? && application.respond_to?(:reload_routes!, true)
-        application.reload_routes!
+      if named_routes.empty?
+        if application.respond_to?(:reload_routes_unless_loaded, true)
+          application.reload_routes_unless_loaded
+        elsif application.respond_to?(:reload_routes!, true)
```

Comment:
Is it even possible for `application` to not respond to this method in supported versions of Rails? Not is the scope of this PR, I would be grateful if you can investigate this too.